File: 2D Array (matrix)/118. Pascal's Triangle/pascal_triangle.java

import java.util.ArrayList;
import java.util.List;

class Solution {

    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> innerList = new ArrayList<>();
        innerList.add(1);
        ans.add(innerList);

        for (int i = 1; i < numRows; i++) {
            List<Integer> temp = new ArrayList<>();
            temp.add(1);

            for (int j = 0; j < ans.get(i - 1).size() - 1; j++) {
                int data = ans.get(i - 1).get(j) + ans.get(i - 1).get(j + 1);
                temp.add(data);
            }
            temp.add(1);
            ans.add(temp);
        }
        return ans;
    }
}

public class pascal_triangle {

    public static void main(String[] args) {
        Solution s = new Solution();
        int numRows = 5;
        List<List<Integer>> ans = s.generate(numRows);

        for (List<Integer> i : ans) {
            System.out.println(i);
        }
    }
}

File: 2D Array (matrix)/118. Pascal's Triangle/pascal_triangle.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back({1});

        for (int i=1; i<numRows; i++){
            vector<int> a;
            a.push_back(1);
            for(int j=0; j<ans[i-1].size()-1; j++){
                int data=ans[i-1][j]+ans[i-1][j+1];
                a.push_back(data);
            }
            a.push_back(1);
            ans.push_back(a);
        }
        return ans;

        
    }
};

int main(){
    Solution Sol;
    int numRows=5;
    vector<vector<int>> ans=Sol.generate(numRows);
    for(int i=0; i<ans.size(); i++){
        // Print leading spaces
        for(int j=0; j<numRows-i-1; j++){
            cout << " ";
        }
        for(int j=0; j<ans[i].size(); j++){
            cout  << ans[i][j] << " ";
        }
        cout << endl;
    }
}